api/clients/v2/examples/client_construction.go

```diff
@@ -0,0 +1,281 @@
+	if err != nil {
+		return nil, fmt.Errorf("create eth reader: %w", err)
+	}
+
+	chainState := eth.NewChainState(ethReader, ethClient)
+	kzgVerifier, err := createKzgVerifier()
+	if err != nil {
+		return nil, fmt.Errorf("create kzg verifier: %w", err)
+	}
+
+	// Create the retrieval client for fetching blobs from DA nodes
+	retrievalClient := clients.NewRetrievalClient(
+		logger,
+		ethReader,
+		chainState,
+		kzgVerifier,
+		10, // Number of concurrent connections to validators
+	)
+
+	// Create the ValidatorPayloadRetriever config
+	validatorPayloadRetrieverConfig := payloadretrieval.ValidatorPayloadRetrieverConfig{
+		PayloadClientConfig: *clients.GetDefaultPayloadClientConfig(),
+		RetrievalTimeout:    30 * time.Second,
+	}
+
+	return payloadretrieval.NewValidatorPayloadRetriever(
+		logger,
+		validatorPayloadRetrieverConfig,
+		retrievalClient,
+		kzgVerifier.Srs.G1)
+}
+
+func createRelayClient(
+	logger logging.Logger,
+	ethClient common.EthClient,
+	relayRegistryAddress gethcommon.Address,
+) (clients.RelayClient, error) {
+	config := &clients.RelayClientConfig{
+		UseSecureGrpcFlag:  true,
+		MaxGRPCMessageSize: 100 * 1024 * 1024, // 100 MB message size limit
+	}
+
+	relayUrlProvider, err := relay.NewRelayUrlProvider(ethClient, relayRegistryAddress)
+	if err != nil {
+		return nil, fmt.Errorf("create relay url provider: %w", err)
+	}
+
+	return clients.NewRelayClient(
+		config,
+		logger,
+		relayUrlProvider)
+}
+
+func createDisperserClient(privateKey string, kzgProver *prover.Prover) (clients.DisperserClient, error) {
+	signer, err := auth.NewLocalBlobRequestSigner(privateKey)
+	if err != nil {
+		return nil, fmt.Errorf("create blob request signer: %w", err)
+	}
+
+	disperserClientConfig := &clients.DisperserClientConfig{
+		Hostname:          "disperser-testnet-holesky.eigenda.xyz",
+		Port:              "443",
+		UseSecureGrpcFlag: true,
+	}
+
+	return clients.NewDisperserClient(
+		disperserClientConfig,
+		signer,
+		kzgProver,
+		nil)
+}
+
+func createKzgVerifier() (*verifier.Verifier, error) {
+	kzgConfig := createKzgConfig()
+	kzgConfig.LoadG2Points = false
+	blobVerifier, err := verifier.NewVerifier(&kzgConfig, nil)
+	if err != nil {
+		return nil, fmt.Errorf("create blob verifier: %w", err)
+	}
+
+	return blobVerifier, nil
+}
+
+func createKzgProver() (*prover.Prover, error) {
+	kzgConfig := createKzgConfig()
+	kzgProver, err := prover.NewProver(&kzgConfig, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return kzgProver, nil
+}
+
+func createCertVerifier() (*verification.CertVerifier, error) {
+	logger, err := createLogger()
+	if err != nil {
+		return nil, fmt.Errorf("create logger: %v", err)
+	}
+
+	ethClient, err := createEthClient(logger)
+	if err != nil {
+		return nil, fmt.Errorf("create eth client: %w", err)
+	}
+
+	return verification.NewCertVerifier(
+		logger,
+		ethClient,
+		verification.NewStaticCertVerifierAddressProvider(
+			gethcommon.HexToAddress("0xFe52fE1940858DCb6e12153E2104aD0fDFbE1162")))
```

Comment:
move to top

Reply:
[0986099d](https://github.com/Layr-Labs/eigenda/pull/1454/commits/0986099d160c376e581ae0235185b1889f2a270b)